### PR TITLE
[v7r0] Just better logging in WMS

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*WMS
FIX: Better logging for WMS Optimizers

ENDRELEASENOTES

### CODE REVIEW

File: WorkloadManagementSystem/Executor/JobScheduling.py [hunk 1]
```diff
@@ -127,19 +132,20 @@ def optimizeJob(self, jid, jobState):
     if checkPlatform and jobPlatform:
       result = gConfig.getOptionsDict('/Resources/Computing/OSCompatibility')
       if not result['OK']:
-        return S_ERROR("Unable to get OSCompatibility list")
+        self.jobLog.error("Unable to get OSCompatibility list")
```
Comment (andresailer) [inline WorkloadManagementSystem/Executor/JobScheduling.py]:
```suggestion
        self.jobLog.error("Unable to get OSCompatibility list", result['Message'])
```

File: WorkloadManagementSystem/Executor/JobScheduling.py [hunk 2]
```diff
@@ -686,6 +691,6 @@ def __checkStageAllowed(self, jobState):
     result = jobState.getAttribute("OwnerGroup")
     if not result['OK']:
       self.jobLog.error("Cannot retrieve OwnerGroup from DB: %s" % result['Message'])
```
Comment (andresailer) [inline WorkloadManagementSystem/Executor/JobScheduling.py]:
```suggestion
       self.jobLog.error("Cannot retrieve OwnerGroup from DB", result['Message'])
```

File: WorkloadManagementSystem/Client/DownloadInputData.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def execute(self, dataToResolve=None):
           if value:
             self.log.verbose('\t%s %s' % (item, value))
 
-    self.log.info('Total size of files to be downloaded is %s bytes' % (totalSize))
+    self.log.info('Total size of files to be downloaded" "is %s bytes' % totalSize)
```
Comment (chrisburr) [inline WorkloadManagementSystem/Client/DownloadInputData.py]:
```suggestion
    self.log.info('Total size of files to be downloaded', 'is %s bytes' % totalSize)
```

File: WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py [hunk 4]
```diff
@@ -5,7 +5,6 @@
 __RCSID__ = "$Id$"
 
 import threading
-import datetime
```
Comment (andresailer) [inline WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py]: Did you verify that this is not needed? The comment below this line refers to this import...
See also: https://github.com/DIRACGrid/DIRAC/pull/3748#discussion_r200631554
If it is not needed that comment should be dropped

File: WorkloadManagementSystem/Executor/JobScheduling.py [hunk 5]
```diff
@@ -431,8 +437,8 @@ def __sendToTQ(self, jobState, jobManifest, sites, bannedSites, onlineSites=None
     else:
       result = jobManifest.createSection(reqSection)
     if not result['OK']:
-      self.jobLog.error("Cannot create %s: %s" % reqSection, result['Value'])
-      return S_ERROR("Cannot create %s in the manifest" % reqSection)
+      self.jobLog.error("Cannot create jobManifest section", "(%s: %s)" % reqSection, result['Value'])
```
Comment (andresailer) [inline WorkloadManagementSystem/Executor/JobScheduling.py]:
```suggestion
      self.jobLog.error("Cannot create jobManifest section", "(%s: %s)" % (reqSection, result['Message']))
```

File: WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py (hunk 4 above)
Comment (fstagni) [reply WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py]: Correct, fixed.

File: WorkloadManagementSystem/Executor/JobScheduling.py (hunk 5 above)
Comment (fstagni) [reply WorkloadManagementSystem/Executor/JobScheduling.py]: Correct, fixed.

**Pull request merged**